src/extension.ts

```diff
@@ -597,3 +610,27 @@ export function _isValidFileForAnalysis(documentUri: vscode.Uri) {
 	return allowedFileTypes.includes(path.extname(documentUri.path));
 }
 
+class VSCodeDisplayable implements Displayable {
```

Comment:
This class can abstract away all of the VSCode-specific stuff about UI/UX, so other classes won't need to interact directly with VSCode as much.